src/core/normalizer.coffee

```diff
@@ -180,11 +179,16 @@ class Normalizer
           value = [parent.getAttribute('class'), value].join(' ')
         parent.setAttribute(name, value)
       dom(node).unwrap()
-      return parent
-    else if node.parentNode.tagName > node.tagName
+      node = parent
+
+    if node.parentNode != root and node.parentNode.tagName > node.tagName
       # Order tag nesting alphabetically (parent->child : A->Z)
-      dom(node).moveChildren(node.parentNode)
-      dom(node.parentNode).wrap(node)
+      dom(node).isolate(root)
+      parent = node.parentNode
+      while parent.parentNode != root and parent.tagName > node.tagName
+        parent = parent.parentNode
+      dom(node).unwrap()
+      dom(parent).wrap(node)
       return node
```

Comment:
This is small, but can't the last three line just be one `return node` at the end of the function?